lib/ansible/modules/cloud/vultr/vultr_server_metal.py

```diff
@@ -0,0 +1,528 @@
+      description: Operating system used for the server
+      returned: success
+      type: str
+      sample: "Debian 9 x64"
+    pending_charges:
+      description: Pending charges
+      returned: success
+      type: float
+      sample: 0.18
+    ram:
+      description: Information about the RAM size
+      returned: success
+      type: str
+      sample: "32768 MB"
+    status:
+      description: Status about the deployment of the server
+      returned: success
+      type: str
+      sample: "active"
+    tag:
+      description: Server tag
+      returned: success
+      type: str
+      sample: "my tag"
+    v6_main_ip:
+      description: Main IPv6
+      returned: success
+      type: str
+      sample: "2001:DB8:9000::100"
+    v6_network:
+      description: Network IPv6
+      returned: success
+      type: str
+      sample: "2001:DB8:9000::"
+    v6_network_size:
+      description:  Network size IPv6
+      returned: success
+      type: int
+      sample: 64
+    v6_networks:
+      description: Networks IPv6
+      returned: success
+      type: list
+      sample: []
+'''
+
+import time
+import base64
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text, to_bytes
+from ansible.module_utils.vultr import (
+    Vultr,
+    vultr_argument_spec,
+)
+
+
+class AnsibleVultrServerMetal(Vultr):
+
+    def __init__(self, module):
+        super(AnsibleVultrServerMetal, self).__init__(module, "vultr_server_metal")
+
+        self.server = None
+        self.returns = {
+            'SUBID': dict(key='id'),
+            'label': dict(key='name'),
+            'date_created': dict(),
+            'allowed_bandwidth_gb': dict(convert_to='int'),
+            'current_bandwidth_gb': dict(),
+            'default_password': dict(),
+            'internal_ip': dict(),
+            'disk': dict(),
+            'cost_per_month': dict(convert_to='float'),
+            'location': dict(key='region'),
+            'main_ip': dict(key='v4_main_ip'),
+            'network_v4': dict(key='v4_network'),
+            'gateway_v4': dict(key='v4_gateway'),
+            'os': dict(),
+            'pending_charges': dict(convert_to='float'),
+            'ram': dict(),
+            'plan': dict(),
+            'status': dict(),
+            'tag': dict(),
+            'v6_main_ip': dict(),
+            'v6_network': dict(),
+            'v6_network_size': dict(),
+            'v6_networks': dict(),
+        }
+        self.server_power_state = None
+
+    def get_startup_script(self):
+        return self.query_resource_by_key(
+            key='name',
+            value=self.module.params.get('startup_script'),
+            resource='startupscript',
+        )
+
+    def get_os(self):
+        return self.query_resource_by_key(
+            key='name',
+            value=self.module.params.get('os'),
+            resource='os',
+            use_cache=True
+        )
+
+    def get_ssh_key(self):
```

Comment:
we recently fixed this in vultr_server to be able to pass multiple keys. See https://github.com/ansible/ansible/pull/50337